core_backend/app/__init__.py

```diff
@@ -73,9 +77,37 @@ def create_app() -> FastAPI:
     metrics_app = create_metrics_app()
     app.mount("/metrics", metrics_app)
 
+    async def reset_quota() -> None:
+        """Reset api daily quota for all users in redis"""
+        redis = app.state.redis
+        async for asession in get_async_session():
+            try:
+                users = await get_all_users(asession)
+                for user in users:
+                    api_daily_quota = encode_api_limit(user.api_daily_quota)
+                    await redis.set(
+                        f"remaining-calls:{user.username}",
+                        api_daily_quota,
+                    )
+                logger.info("Successfully Synced Redis to database")
+            except Exception as e:
+                logger.info(f"Redis sync failed: {e}")
+
     @app.on_event("startup")
-    def startup_event() -> None:
+    async def startup_event() -> None:
         """Startup event"""
+        app.state.redis = await aioredis.from_url(redis_host)
         logger.info("Application started")
+        scheduler = AsyncIOScheduler()
+        await reset_quota()
+        scheduler.add_job(reset_quota, "interval", minutes=1440)
```

Comment:
Does this mean it resets at 00:00:00 UTC?

Reply:
Should this be based on a time / calendar trigger instead of interval?

Reply:
Also, ideally it would be X calls in a 24 hrs period. So some sort of rolling limit. But I can see that might be tricky so ok to park it if it's not simple.